scripts/lib/CIME/case.py

```diff
@@ -443,13 +443,11 @@ def _set_compset_and_pesfile(self, compset_name, files, user_compset=False, pesf
                     user_mods_dir     = files.get_value("USER_MODS_DIR"     , {"component":component}, resolved=False)
                     self.set_lookup_value("COMPSETS_SPEC_FILE" ,
                                    files.get_value("COMPSETS_SPEC_FILE", {"component":component}, resolved=False))
+                    self._principle_component = component.upper()
```

Comment:
This determination of principle component will not work for a user-defined compset. I feel this is very important given that we encourage people to use compset long names rather than aliases. And, with @mvertens 's idea in #608 we'll need a way to determine the principle component for a user-defined compset for that purpose, too.

If it's critical to get this PR in without support for user-defined compsets, I'm okay with that - with a separate issue being opened that this needs to be fixed.